BuildingBlocks/src/BuildingBlocks.Infrastructure/Persistence/BlobStorage/AzureStorageAccount/AzureStorageAccount.cs

```diff
@@ -95,10 +100,15 @@ private async Task UploadChangedBlobs()
         var changedBlobs = new Dictionary<BlobClient, byte[]>(_changedBlobs);
         foreach (var (cloudBlockBlob, bytes) in changedBlobs)
         {
-            await using var memoryStream = new MemoryStream(bytes);
+            var memoryStream = await _logger.TraceTime(async () => new MemoryStream(bytes), nameof(MemoryStream));
```

Comment:
No need to log this call. It shouldn't even take 1ms.

Also, you removed the `using` as part of your changes, which isn't good.

Same for the other blob storage implementations.